File: frontend-web/src/types/member.ts
export type Member = {
  id: string;
  authId: string;
  firstName: string;
  secondName: string;
  phone: string;
  birthDate: string;
  objective: string;
  medicalRestrictions: string;
  nfcCardId: string | null;
  nfcActive: boolean;
  suspended: boolean;
  gender: "MALE" | "FEMALE";
  profilePic: string | null;
  subscriptionPlan: string | null;
  subscriptionStatus: string | null;
};

export type MembersResponse = {
  content: Member[];
  page: number;
  size: number;
  totalElements: number;
  totalPages: number;
  hasMore: boolean;
};

export type CreateMemberPayload = {
  firstName: string;
  secondName: string;
  phone: string;
  birthDate: string;
  gender: "MALE" | "FEMALE";
  objective: string;
  medicalRestrictions: string;
};

export type UpdateMemberPayload = {
  firstName?: string;
  secondName?: string;
  phone?: string;
  birthDate?: string;
  gender?: "MALE" | "FEMALE";
  objective?: string;
  medicalRestrictions?: string;
  subscriptionPlan?: string | null;
  subscriptionStatus?: string | null;
};

File: backend/user-service/src/main/java/com/gym/userservice/model/Member.java
package com.gym.userservice.model;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "members")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Member {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String authId;

    @Column(unique = true)
    private String email;

    @Column(nullable = true)
    private String firstName;

    @Column(nullable = true)
    private String secondName;

    private String phone;

    private String birthDate;

    private String objective;

    private String medicalRestrictions;

    @Column(unique = true)
    private String nfcCardId;

    @Column(nullable = false)
    private boolean nfcActive = false;

    @Column(nullable = false)
    private boolean suspended = false;

    @Column
    private String gender;

    @Column
    private String profilePic;

    @Column
    private String subscriptionPlan;

    @Column
    private String subscriptionStatus;
}